### DISCUSSION TITLE
The .subs() function is giving an error.

### DISCUSSION THREAD

**Discussion by ItshMoh** [sympy/sympy#25828] [Q&A]
I was performing a simple operation on sympy. I think the answer given by sympy is incorrect.
the code is 
```
expr = x**4 - 4*x**3 + 4*x**2 - 2*x + 3
replacements = [(x**i, y**i) for i in range(3) if i % 2 == 0]
expr.subs(replacements)
```

output by sympy is    −4*x**3 -2*x +y**4 +4*x**2 +3

Real output should be  −4*x**3 -2*x +x**4 +4*y**2 +3

**Comment by 1e9abhi1e10** [comment]
You are using the `.subs()` method to substitute the expressions in `expr` with the values from `replacements`. 
replacements  = `[(1, 1), (x**2, y**2)]`
The resulting expression will have `x**0` replaced by `1`, and `x**2` replaced by `y**2`. The final expression will be:

`-4*x**3 - 2*x + y**4 + 4*y**2 + 3`

**Reply by ItshMoh** [reply]
Sorry for the mistake in real output. 
I have understood how term 4*y**2 has come but I have the doubt why x**4 changes to y**4 as (i) is just running till 3. So x**4 should not be updated.

**Comment by 1e9abhi1e10** [comment]
> but I have the doubt why x**4 changes to y**4 as (i) is just running till 3. So x**4 should not be updated.

`x**2` is replaced by `y**2`, so `x**4` becomes `y**4`.
